src/utils/traverseHeadings.ts

```diff
@@ -8,64 +11,92 @@ export function traverseHeadings(tree, filterKey: string): string[] {
+    if (typeof node !== 'object') continue;
+    if (!('props' in node)) continue;
 
-    if ("fragments" in node.props) {
+    if ('fragments' in node.props) {
       // Recurse on the fragment corresponding to this page's filterKey
       if (filterKey in node.props.fragments) {
         const fragmentFunction = node.props.fragments[filterKey];
         const fragment = fragmentFunction([]); // expand function into full tree
         headings = headings.concat(traverseHeadings(fragment, filterKey));
-      } else if ("all" in node.props.fragments) {
+      } else if ('all' in node.props.fragments) {
         // "all" includes every filterKey, so recurse
         const fragmentFunction = node.props.fragments.all;
         const fragment = fragmentFunction([]); // expand function into full tree
+
         headings = headings.concat(traverseHeadings(fragment, filterKey));
       }
-    } else if ("children" in node.props) {
+    } else if ('children' in node.props) {
       // Recurse on the children, _unless_ this is a FilterContent with a
       // filter that doesn't match the current filterKey
-      if (node.props.mdxType === "FilterContent") {
+      if (node.props.mdxType === 'FilterContent') {
         let filterContentFilter;
-        if ("framework" in node.props)
+        if ('framework' in node.props)
           filterContentFilter = node.props.framework;
-        if ("integration" in node.props)
+        if ('integration' in node.props)
           filterContentFilter = node.props.integration;
-        if ("platform" in node.props) filterContentFilter = node.props.platform;
+        if ('platform' in node.props) filterContentFilter = node.props.platform;
         if (filterContentFilter === filterKey) {
           headings = headings.concat(
-            traverseHeadings(node.props.children, filterKey),
+            traverseHeadings(node.props.children, filterKey)
           );
         }
       } else {
         headings = headings.concat(
-          traverseHeadings(node.props.children, filterKey),
+          traverseHeadings(node.props.children, filterKey)
         );
       }
 
       // Is this a heading?  If so, "children" is actually the heading text
-      if ("mdxType" in node.props) {
+      if ('mdxType' in node.props) {
         const mdxType = node.props.mdxType;
-        if (mdxType === "h2" || mdxType === "h3") {
+
+        if (mdxType === 'h2' || mdxType === 'h3') {
           headings.push([node.props.children, node.props.id, mdxType]);
         }
+        if (mdxType === 'Accordion') {
+          const id = node.props.title.replace(/\s+/g, '-').toLowerCase();
+          const type = node.props.headingLevel
+            ? 'h' + node.props.headingLevel
+            : 'div';
+          if (type === 'h2' || type === 'h3') {
+            headings.push([node.props.title, id, type]);
+          }
+        }
+
+        // Do not include headings from within an Accordion in headings array
+        if (mdxType === 'wrapper') {
```

Comment:
What is the `wrapper` mdxType?  Also it looks like we are looking for a very specific structure and then trying to remove and headings we find from the heading array.  Are we guaranteed to have this specific structure?  Is there a way we can detect headings don't belong in the array before adding them in, instead of removing them afterwards?